x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -972,6 +978,8 @@ private static Expression processScalarFunction(EsqlScalarFunction f, EsqlFuncti
             boolean childrenChanged = false;
             DataType targetDataType = DataType.NULL;
             Expression arg;
+            DataType targetNumericType = DataType.NULL;
+            boolean castNumericArgs = false;
             for (int i = 0; i < args.size(); i++) {
```

Comment:
I personally find this `for` loop a bit more explicit in its intent in the following way.

```
            for (int i = 0; i < args.size(); i++) {
                arg = args.get(i);
                var dataType = arg.dataType();

                if (arg.resolved()) {
                    if (dataType == KEYWORD) {
                        if (arg.foldable() && ((arg instanceof EsqlScalarFunction) == false)) {
                            if (i < targetDataTypes.size()) {
                                targetDataType = targetDataTypes.get(i);
                            }
                            if (targetDataType != DataType.NULL && targetDataType != DataType.UNSUPPORTED) {
                                Expression e = castStringLiteral(arg, targetDataType);
                                childrenChanged = true;
                                newChildren.add(e);
                                continue;
                            }
                        }
                    } else if (dataType.isNumeric() && canCastMixedNumericTypes(f)) {
                        if (targetNumericType.isNumeric() == false) {
                            targetNumericType = dataType;  // target data type is the first numeric data type
                        } else if (dataType != targetNumericType) {
                            if (canCastNumeric(dataType, targetNumericType)) {
                                castNumericArgs = true;
                            } else {
                                castNumericArgs = false;
                            }
                        }
                    }
                }
                newChildren.add(args.get(i));
            }
```